osmnx/pois.py

```diff
@@ -0,0 +1,414 @@
+            vertices[result['id']] = {'lat': result['lat'],
+                                      'lon': result['lon']}
+    return vertices
+
+def parse_osm_way(vertices, response):
+    """
+    Parse ways (areas) from OSM node vertices.
+
+    Parameters
+    ==========
+    vertices : Python dict 
+        Node vertices parsed from OSM response.  
+
+    Returns
+    =======
+    Dict of vertice IDs and their lat, lon coordinates.
+    """
+    if 'type' in response and response['type'] == 'way':
+        nodes = response['nodes']
+        try:
+            polygon = Polygon([(vertices[node]['lon'], vertices[node]['lat']) for node in nodes])
+
+            poi = {'nodes': nodes,
+                   'geometry': polygon,
+                   'osmid': response['id']}
+
+            if 'tags' in response:
+                for tag in response['tags']:
+                    poi[tag] = response['tags'][tag]
+            return poi
+
+        except Exception:
+            log('Polygon has invalid geometry: {}'.format(nodes))
+    return None
+
+def parse_osm_node(response):
+    """
+    Parse points from OSM nodes.
+
+    Parameters
+    ==========
+    response : JSON 
+        Nodes from OSM response.  
+
+    Returns
+    =======
+    Dict of vertice IDs and their lat, lon coordinates.
+    """
+    try:
+        point = Point(response['lon'], response['lat'])
+
+        poi = {
+            'osmid': response['id'],
+            'geometry': point
+        }
+        if 'tags' in response:
+            for tag in response['tags']:
+                poi[tag] = response['tags'][tag]
+    except Exception:
+        log('Point has invalid geometry: {}'.format(response['id']))
+
+    return poi
+
+def invalid_multipoly_handler(gdf, relation, way_ids):
+    """
+    Handles invalid multipolygon geometries when there exists e.g. a feature without geometry (geometry == NaN)
+
+    Parameters
+    ==========
+
+    gdf : gpd.GeoDataFrame
+        GeoDataFrame with Polygon geometries that should be converted into a MultiPolygon object.
+    relation : dict
+        OSM 'relation' dictionary
+    way_ids : list
+        A list of 'way' ids that should be converted into a MultiPolygon object. 
+    """
+    try:
+        gdf_clean = gdf.dropna(subset=['geometry'])
+        multipoly = MultiPolygon(list(gdf_clean['geometry']))
+        return multipoly
+    except Exception:
+        log("Invalid geometry at relation id %s.\nWay-ids of the invalid MultiPolygon:" % (
+        relation['id'], str(way_ids)))
+        return None
+
+def parse_osm_relations(relations, osm_way_df):
+    """
+    Parses the osm relations from osm ways and nodes.
+         
+    Parameters
+    ==========
+    relations : list
+        OSM 'relation' items (dictionaries) in a list. 
+    osm_way_df : gpd.GeoDataFrame
+        OSM 'way' features as a GeoDataFrame that contains all the 'way' features that will constitute the multipolygon relations.
+     
+    Information
```

Comment:
Can we remove the "information" header and move the URL into the top of the docstring with the sentence description?

Reply:
Yes a good idea. Done.